src/Middleware/JsonValidator.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Psr7Middlewares\Middleware;
+
+use JsonSchema\Validator;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+
+class JsonValidator
+{
+    /** @var \stdClass */
+    private $schema;
+
+    /**
+     * JsonSchema constructor.
+     *
+     * @param \stdClass $schema A JSON-decoded object-representation of the schema.
+     */
+    public function __construct(\stdClass $schema)
+    {
+        $this->schema = $schema;
+    }
+
+    /**
+     * Execute the middleware.
+     *
+     * @param ServerRequestInterface $request
+     * @param ResponseInterface $response
+     * @param callable $next
+     *
+     * @return ResponseInterface
+     * @throws \RuntimeException
+     * @throws \InvalidArgumentException
+     * @throws \JsonSchema\Exception\ExceptionInterface
+     */
+    public function __invoke(ServerRequestInterface $request, ResponseInterface $response, callable $next)
+    {
+        $value = $request->getParsedBody();
+        if (!is_object($value)) {
+            return $this->invalidateResponse(
+                $response,
+                sprintf('Parsed body must be an object. Type %s is invalid.', gettype($value))
+            );
+        }
+
+        $validator = new Validator();
+        $validator->check($value, $this->schema);
+
+        if (!$validator->isValid()) {
+            return $this->invalidateResponse(
+                $response,
+                'Unprocessable Entity',
+                [
+                    'Content-Type' => 'application/json',
+                ],
+                json_encode($validator->getErrors(), JSON_UNESCAPED_SLASHES)
```

Comment:
Maybe the response generation can be configured in a similar way than errorHandler or shutdown, providing a callable handler:

```php
$errorResponse = function ($request, $response) {
    $validator = JsonValidator::getValidator($request);
    $response = $response->withHeader('Content-Type', 'application/json');
    $response->getBody()->write(json_encode($validator->getErrors(), JSON_UNESCAPED_SLASHES));
};

$middlewares[] = Middlewares::JsonValidator()->errorHandler($errorResponse);
```
And if no custom errorHandler is provided, use the default.